components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/AbstractRequestBuilder.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth2;
+
+import org.wso2.carbon.identity.base.IdentityException;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * This is a global level interface for building requests.
+ */
+public interface AbstractRequestBuilder {
```

Comment:
Lets remove the "Abstract" part form the class name. I actually thought this was a abstract class looking at code references. Also shall we add more context to the class name such as "OAuthAuthoritzationRequestBuilder" or "AuthoritzationRequestBuilder" without simply saying ...RequestBuilder as the system deals with multiple types of requests it can be confusing to implementers